onnxruntime/core/providers/tvm/tvm_api.cc

```diff
@@ -108,20 +101,19 @@ void TVM_VM_GetOutputs(TvmModule& mod,
 }
 
 void TVMGetOutputShapes(TvmModule& mod,
-                        size_t num_outputs,
-                        std::vector<std::vector<int64_t>>& output_shapes)
+                        TVMTensorShapes& output_shapes)
 {
-  output_shapes.clear();
+  size_t size = output_shapes.size();
   TvmPackedFunc get_output = mod.GetFunction("get_output", false);
-  for (size_t i = 0; i < num_outputs; ++i) {
+  for (size_t i = 0; i < size; ++i) {
     ::tvm::runtime::NDArray output_array = get_output(i);
```

Comment:
Any change this goes outside the boundaries?

Reply:
I'm not sure that understand your question. But if you mean that size can be incorrect it means that there is inconsistent between TVM and ORT due to size is number of outputs expected by ORT and extracted from onnx model, TVM use the same model and should return the same outputs number (moreover in the same order with expected shapes). Otherwise it is issue of interaction ORT and TVM. I agree that it is not strong place, I will think and try to validate size value in the future development and PR.